plugins/org.jkiss.dbeaver.ext.clickhouse/src/org/jkiss/dbeaver/ext/clickhouse/model/ClickhouseSQLDialect.java

```diff
@@ -104,4 +109,17 @@ public String getColumnTypeModifiers(@NotNull DBPDataSource dataSource, @NotNull
     public boolean supportsNestedComments() {
         return true;
     }
+
+    //We should quote keywords which is not keywords for clickhouse, otherwise JSQLParser can't parse statements
+    @Override
+    public boolean mustBeQuoted(@NotNull String str, boolean forceCaseSensitive) {
+        boolean isSpecialWord = false;
+        for (String word : CLICKHOUSE_NONKEYWORDS) {
+            if (word.equalsIgnoreCase(str)) {
+                isSpecialWord = true;
+                break;
+            }
+        }
+        return isSpecialWord || super.mustBeQuoted(str, forceCaseSensitive);
```

Comment:
And there it would become a simple delegate to the super method:

```suggestion
        return super.mustBeQuoted(str, forceCaseSensitive);
```